src/world/mod.rs

```diff
@@ -16,11 +16,49 @@ mod lazy;
 #[cfg(test)]
 mod tests;
 
+#[cfg(not(feature = "nightly"))]
 const COMPONENT_NOT_REGISTERED: &str = "No component with the given id. Did you forget to register \
                                         the component with `World::register::<ComponentName>()`?";
+#[cfg(not(feature = "nightly"))]
 const RESOURCE_NOT_ADDED: &str = "No resource with the given id. Did you forget to add \
                                   the resource with `World::add_resource(resource)`?";
 
+#[cfg(feature = "nightly")]
+fn nightly_component_message<T>(id: usize) -> String {
+    let type_name = unsafe { ::std::intrinsics::type_name::<T>() };
+    let (message, function) = if id == 0 {
```

Comment:
Hm, this formatting looks weird

Reply:
Not sure how I could make it better without making the lines super long though. `cargo +nightly fmt` doesn't seem to do anything to it either.